### PR TITLE
logrotate.service: harden with ProtectHostname and RestrictSUIDSGID

### PR DESCRIPTION
**Pull request by cgzones**
Prohibit changing the system hostname/domain and creating SUID/SGID files

### CODE REVIEW

File: examples/logrotate.service [hunk 1]
```diff
@@ -24,9 +24,11 @@ PrivateDevices=true
...
 ProtectClock=true
 ProtectControlGroups=true
+ProtectHostname=true
 ProtectKernelLogs=true
 ProtectKernelModules=true
 ProtectKernelTunables=true
 ProtectSystem=full
 RestrictNamespaces=true
 RestrictRealtime=true
+RestrictSUIDSGID=true
```
Comment (kdudka) [inline examples/logrotate.service:34]: I am not sure about this one.  Setting setuid/setgid bits on directories is quite common.  What is the motivation behind this change?

Comment (cgzones) [reply examples/logrotate.service:34]: Is it really common to set setuid/setgid bits inside of logrotate scripts?

Comment (kdudka) [reply examples/logrotate.service:34]: Who are you asking here?  Users of logrotate?  This is not the channel to reach them...

Note that these bits are not limited to scripts only.  They can be set by logrotate directly, for example: `createolddir 6770`

Comment (cgzones) [reply examples/logrotate.service]: OK, I understand now.
Converted into a comment why not using RestrictSUIDSGID

**Comment by kdudka** [comment]
Merging, thanks!
